code/modules/mining/lavaland/megafauna_loot.dm

```diff
@@ -601,6 +601,11 @@
 	animate(src)
 	SpinAnimation(15)
 
+/obj/item/soulscythe/Destroy(force)
+	. = ..()
+	soul.ghostize()
+	QDEL_NULL(soul)
+
```

Comment:
Unless things have changed in my long winter, you generally want to clean up the contents of an object _before_ you call parent `Destroy()`.
```suggestion
/obj/item/soulscythe/Destroy(force)
	soul.ghostize()
	QDEL_NULL(soul)
	. = ..()
```